File: catalog-prod/src/main/java/com/machado/catalog_prod/entity/Product.java
package com.machado.catalog_prod.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @NotNull
    @Size(max = 255)
    private String name;

    private String description;

    @NotNull
    @Positive
    private Double price;

    @NotNull
    @ManyToOne(cascade = CascadeType.ALL)
    private Category category;
}

File: catalog-client/src/app/models/product.ts
import { Category } from "./category";

export class Product {

  id!: number;
  name!: string;
  description!: string;
  price!: number;
  category!: Category

  constructor(id: number, name: string, description: string, price: number, category: Category){
    this.id = id;
    this.name = name
    this.description = description;
    this.price = price;
    this.category = category;
  }

}
